python/nav/django/auth.py

```diff
@@ -45,10 +47,19 @@
 class AuthenticationMiddleware(MiddlewareMixin):
     def process_request(self, request):
         session = request.session
-
-        if ACCOUNT_ID_VAR not in session:
-            session[ACCOUNT_ID_VAR] = Account.DEFAULT_ACCOUNT
-        account = Account.objects.get(id=session[ACCOUNT_ID_VAR])
+        account = None
+
+        if ACCOUNT_ID_VAR not in session:  # Not logged in
```

Comment:
This does not signify merely "Not logged in". It signifies that that this is a *new* session, and no Account is associated with it yet. This is why the original associates the anonymous `Account.DEFAULT_ACCOUNT` with the session in this case.

**However**, when we reach this test, the session may already exist and therefore already be associated with the anonymous user. The way this logic is structured now, if a user:

1. Browses NAV anonymously (if enabled). A new session is created and a cookie stored in their browser.
2. Logs in through whatever federated mechanism the web server is configured with
3. The user browses NAV again, expecting to be authenticated - but since they already have an active NAV session, they now need to delete their NAV cookie to make NAV able to recognize them.

Reply:
This sounds like a question for Pettai. What happens if you're already logged in with some user, and REMOTE_USER is set by something else to some other user than who you already are. How do we still support the magic admin-user for instance.

Reply:
Currently only "not logged in at all" is supported.